java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -1708,6 +1724,32 @@ private void nextVector(Decimal64ColumnVector result,
       result.scale = (short) scale;
     }
 
+    private void setIsRepeatingIfNeeded(Decimal64ColumnVector result, int index) {
+      if (result.isRepeating
+          && index > 0
+          && (result.vector[0] != result.vector[index] ||
+          result.isNull[0] != result.isNull[index])) {
+        result.isRepeating = false;
+      }
+    }
+
+    private void setIsRepeatingIfNeeded(DecimalColumnVector result, int index) {
+      if (result.isRepeating
+          && index > 0
+          && (!eq(result.vector[0], result.vector[index]) ||
+          result.isNull[0] != result.isNull[index])) {
+        result.isRepeating = false;
+      }
+    }
+
+    private boolean eq(HiveDecimalWritable first, HiveDecimalWritable other) {
```

Comment:
Is there a reason why we cannot use `HiveDecimalWritable.equals` here?
- https://github.com/apache/hive/blob/293a448296933b7498a91e7eeb91edc88dfaa07e/storage-api/src/java/org/apache/hadoop/hive/serde2/io/HiveDecimalWritable.java#L1003-L1019
```java
  public boolean equals(Object other) {
    if (!isSet) {
      return false;
    }
    if (this == other) {
      return true;
    }
    if (other == null || getClass() != other.getClass()) {
      return false;
    }
    HiveDecimalWritable otherHiveDecWritable = (HiveDecimalWritable) other;
    if (!otherHiveDecWritable.isSet()) {
      return false;
    }
    return fastEquals((FastHiveDecimal) otherHiveDecWritable);

  }
```